### PR TITLE
Rename poke-filter-popover, extract template (still inline thanks to …

### PR DESCRIPTION
**Pull request by MajorBreakfast**
…webpack)

### CODE REVIEW

File: ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts [hunk 1]
```diff
@@ -8,19 +8,8 @@ import { PokeFilterPokemonTabComponent } from '../poke-filter-pokemon-tab/poke-f
...
-        <ion-segment-button value="time">Time</ion-segment-button>
-        <ion-segment-button value="pokemon">Pokemon</ion-segment-button>
-      </ion-segment>
-    </ion-toolbar>
-    <div [ngSwitch]="currentTab">
-      <poke-filter-time-tab *ngSwitchCase="'time'"></poke-filter-time-tab>
-      <poke-filter-pokemon-tab *ngSwitchCase="'pokemon'"></poke-filter-pokemon-tab>
-    </div>
-    `,
+  selector: 'poke-filter-popover',
```
Comment (philbu) [inline ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts:11]: Why the selector?

File: ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts [hunk 2]
```diff
@@ -32,7 +21,7 @@ import { FilterService } from '../../services/filter.service';
   ],
 })
 
-export class FilterPopoverComponent {
+export class PokeFilterPopoverComponent {
```
Comment (WoH) [inline ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts:24]: No. It's good practice to prefix the selectors, but Prefixing Component names was a bad idea in the first place. I'd rather rename the other Poke-named Components.

**Review by WoH — CHANGES_REQUESTED** [review]

File: ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts (hunk 2 above)
Comment (MajorBreakfast) [reply ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts:24]: The tag name/selector should mirror the class name. Tags cannot be scoped so prefixing them makes sense

File: ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts (hunk 1 above)
Comment (MajorBreakfast) [reply ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts:11]: So that it is properly named in the dom tree

**Comment by mfkaptan** [comment]
Looks good to me

**Comment by MajorBreakfast** [comment]
I'm merging this now.

@WoH To address your concern: My response was  "The tag name/selector should mirror the class name. Tags cannot be scoped so prefixing them makes sense". I'm merging this due to there being a lot of classes in the project that are named my way. Nevertheless I can be swayed on that. But then we should consistently rename all of them. I won't do that myself because I like the names how they are right now. So feel free to rename them as you feel fit.

File: ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts (hunk 2 above)
Comment (WoH) [reply ionic2/app/components/poke-filter-popover/poke-filter-popover.component.ts:24]: "The tag name/selector should mirror the class name." Wrong.

https://angular.io/styleguide

**Comment by WoH** [comment]
I am requesting changes for good reason.

**Comment by WoH** [comment]
Wtf is this?

**Comment by MajorBreakfast** [comment]
See comment above

**Comment by WoH** [comment]
Yeah, just because a lot of things are done suboptimally, we should 100% extend bad practices all over the app. I agree.

**Comment by gyachdav** [comment]
boyz plz chill the language

**Comment by MajorBreakfast** [comment]
@gyachdav It's a bit heated but we didn't swear ;)

@WoH I agree. We should name things consistently and I propose to prefix all others. I can provide a PR for that. Shall I?

**Comment by WoH** [comment]
My bad, I was thrown off by someone merging a PR despite my change requests.

**Comment by WoH** [comment]
I am working on it.

**Comment by MajorBreakfast** [comment]
Prefix selectors with `poke-` and remove prefixes from classes then? I don't like it but I'm fine with it if it is consistent.

**Comment by MajorBreakfast** [comment]
We should make a folder for each component while we're at it. The `poke-details/` folder currently holds a whole bunch of components

**Comment by WoH** [comment]
Already done, this was really bugging me aswell.
Just like pipes and models in component subfolders.

**Comment by MajorBreakfast** [comment]
For reference to for anyone who reads this in the future: https://github.com/PokemonGoers/Catch-em-all/pull/112 (PR for naming conventions)

**Pull request merged**